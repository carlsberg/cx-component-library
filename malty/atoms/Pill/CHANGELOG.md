# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 0.4.0 (2024-05-31)

### Bug Fixes

- **MS-110:** Pill flickering ([#386](https://github.com/CarlsbergGBS/cx-component-library/issues/386)) ([41674e0](https://github.com/CarlsbergGBS/cx-component-library/commit/41674e022364644f1949eb1fb9b642ad7a4c5fc6))
- **pill:** [DSM-957] rounded in all sizes ([#866](https://github.com/CarlsbergGBS/cx-component-library/issues/866)) ([329ca8a](https://github.com/CarlsbergGBS/cx-component-library/commit/329ca8ac798d0264cc6a6a558da118b9630109d8))
- **pills:** [DSM-988] updated pills text color ([#886](https://github.com/CarlsbergGBS/cx-component-library/issues/886)) ([4fc077e](https://github.com/CarlsbergGBS/cx-component-library/commit/4fc077e3dc99c51ad581ab31a8a41c5b7e37abfd))
- Remove typography provider from every component ([#383](https://github.com/CarlsbergGBS/cx-component-library/issues/383)) ([5d05680](https://github.com/CarlsbergGBS/cx-component-library/commit/5d05680e40d3da5bb3042bd8191427a70606afc3))
- reverted changes ([#396](https://github.com/CarlsbergGBS/cx-component-library/issues/396)) ([ecfcf57](https://github.com/CarlsbergGBS/cx-component-library/commit/ecfcf573863253fea16e24764b6f452c3cc4f215))
- Run auto fixable lint based on new `eslint` configuration ([#501](https://github.com/CarlsbergGBS/cx-component-library/issues/501)) ([9ee94cc](https://github.com/CarlsbergGBS/cx-component-library/commit/9ee94cc2cd407f717e62d8857b5f20a74e7bbec4))

### Features

- add tsconfig files and update package.json files for each component ([3bfc644](https://github.com/CarlsbergGBS/cx-component-library/commit/3bfc644e1cfc9dbb7cf7a0469e25fce055b53240))
- change organization name ([a0aa428](https://github.com/CarlsbergGBS/cx-component-library/commit/a0aa428f43138af5707a3ff4c0e36b7c056f02e6))
- **DSM-228:** Added rating component ([#379](https://github.com/CarlsbergGBS/cx-component-library/issues/379)) ([ab47354](https://github.com/CarlsbergGBS/cx-component-library/commit/ab47354db803fb051a1f87d66ee34966698df8bd))
- **DSM-248:** Headline HTML tags ([#184](https://github.com/CarlsbergGBS/cx-component-library/issues/184)) ([305f5e0](https://github.com/CarlsbergGBS/cx-component-library/commit/305f5e0fa92d5813f4877b338ad827a8c347d452))
- **DSM-362:** snapshots removed & unit test updates ([#307](https://github.com/CarlsbergGBS/cx-component-library/issues/307)) ([78423eb](https://github.com/CarlsbergGBS/cx-component-library/commit/78423eb5cfe04fe8a3a7af1f5af074ac6d35e235))
- **DSM-439:** Change font-family behavior ([#470](https://github.com/CarlsbergGBS/cx-component-library/issues/470)) ([ab6dad4](https://github.com/CarlsbergGBS/cx-component-library/commit/ab6dad414c42c4343de3de6abaa4cd3ff98cfca1))
- **general:** [DSM-953] refactor icons implementation ([#892](https://github.com/CarlsbergGBS/cx-component-library/issues/892)) ([08d259f](https://github.com/CarlsbergGBS/cx-component-library/commit/08d259f476ce97e9db8ac41d5eb4773eabca0d37))
- **global:** [DSM-784] add space props in all components ([#842](https://github.com/CarlsbergGBS/cx-component-library/issues/842)) ([9f03f59](https://github.com/CarlsbergGBS/cx-component-library/commit/9f03f59e7518a99abcea06bacbee989d1ddb0465))
- **global:** migrate from bit to lerna ([a0110c8](https://github.com/CarlsbergGBS/cx-component-library/commit/a0110c8831370dc762c193b17cc593eed381f990))
- **global:** update storybook files to be compliant with v7 ([#970](https://github.com/CarlsbergGBS/cx-component-library/issues/970)) ([c61d4b4](https://github.com/CarlsbergGBS/cx-component-library/commit/c61d4b40a0755becf942ad3f28758a159f8c54e4))
- new icon colors ([#215](https://github.com/CarlsbergGBS/cx-component-library/issues/215)) ([9247d33](https://github.com/CarlsbergGBS/cx-component-library/commit/9247d3370bcde78432033ba8b23837f97e0b0dc0))
- Pill component spacing fix ([#144](https://github.com/CarlsbergGBS/cx-component-library/issues/144)) ([ca53568](https://github.com/CarlsbergGBS/cx-component-library/commit/ca535683aad08c2078f217c186cc40a1227a9b31))
- **Pill:** [DSM-568] Update Pill to match Badge behaviour ([#580](https://github.com/CarlsbergGBS/cx-component-library/issues/580)) ([e513565](https://github.com/CarlsbergGBS/cx-component-library/commit/e5135656d392c66346e0a2c1caf21c4fe93e0961))
- **pill:** [DSM-569] update text font size for the xs variation ([#724](https://github.com/CarlsbergGBS/cx-component-library/issues/724)) ([2a19c3d](https://github.com/CarlsbergGBS/cx-component-library/commit/2a19c3d31f550e67d8aafaac288074b24181962a))
- **pill:** [DSM-744] update color options ([#730](https://github.com/CarlsbergGBS/cx-component-library/issues/730)) ([179d23d](https://github.com/CarlsbergGBS/cx-component-library/commit/179d23d121a9e93fe131f2aa23d505d9410893a3))
- **pill:** [DSM-810] allow uppercase text control via css ([#791](https://github.com/CarlsbergGBS/cx-component-library/issues/791)) ([f4b592f](https://github.com/CarlsbergGBS/cx-component-library/commit/f4b592f591bd1d9120f62b5e6049c3fe644db357))
- **pill:** [DSM-831] add disable state ([#815](https://github.com/CarlsbergGBS/cx-component-library/issues/815)) ([8421ba4](https://github.com/CarlsbergGBS/cx-component-library/commit/8421ba410004b904b92d0f3f610c77d00c0c1edd))
- **pill:** [DSM-920] add tests ([#874](https://github.com/CarlsbergGBS/cx-component-library/issues/874)) ([f469d20](https://github.com/CarlsbergGBS/cx-component-library/commit/f469d20f8ddce0dd4204b0dd756ba5161b1c7cd7))
- remove empty composition files ([f8173f4](https://github.com/CarlsbergGBS/cx-component-library/commit/f8173f4a2ecbf80bb7b6ffe848c023ae31819c2d))
- Several component updates ([#185](https://github.com/CarlsbergGBS/cx-component-library/issues/185)) ([633c9f0](https://github.com/CarlsbergGBS/cx-component-library/commit/633c9f0a4432e9e3d08ff05156453660929c19a2))
- Use new `eslint-config-carlsberg` with streamlined way to create components ([#559](https://github.com/CarlsbergGBS/cx-component-library/issues/559)) ([a459332](https://github.com/CarlsbergGBS/cx-component-library/commit/a45933215bc0b523220743f17d988d7f1ad5fce6))
