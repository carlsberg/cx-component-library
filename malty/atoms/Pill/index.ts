export { Pill } from './Pill';
export { PillColor, PillSize } from './Pill.types';
export type { PillProps } from './Pill.types';
