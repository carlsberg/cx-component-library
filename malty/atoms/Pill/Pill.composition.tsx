import React from 'react';
import { Pill } from './Pill';

export const BasicPill = () => <Pill text="Pill text" />;
