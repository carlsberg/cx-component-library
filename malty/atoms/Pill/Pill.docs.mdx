---
labels: ['react', 'typescript', 'icon']
description: 'A Icon component.'
---

import { MaltyThemeProvider } from '@carlsberggroup/malty.theme.malty-theme-provider';
import { Pill } from './Pill';

Pill example:

A simple Icon component with some text

Live Playground:

```js live
<MaltyThemeProvider theme="global">
  <Pill text="Pill text" />
</MaltyThemeProvider>
```
