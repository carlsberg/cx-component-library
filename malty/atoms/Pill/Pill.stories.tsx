import { CarlsbergFilled } from '@carlsberggroup/malty.icons.carlsberg-filled';
import { allIconsStoryOptions } from '@carlsberggroup/malty.utils.all-icons';
import { generateStorybookSpacing } from '@carlsberggroup/malty.utils.space';
import { Meta, StoryObj } from '@storybook/react';
import React from 'react';
import { Pill } from './Pill';
import { PillProps, PillSize, PillType } from './Pill.types';

const meta: Meta<PillProps> = {
  title: 'Information/Pill',
  component: Pill,
  parameters: {
    importObject: 'Pill',
    importPath: '@carlsberggroup/malty.atoms.pill'
  },
  argTypes: {
    text: {
      control: 'text',
      description: 'Pill content text'
    },
    size: {
      description: 'Pill size, options below',
      options: Object.values(PillSize),
      control: 'select',
      table: {
        category: 'Styling',
        defaultValue: {
          summary: PillSize.Medium
        }
      }
    },
    type: {
      description: 'Pill type',
      options: Object.keys(PillType),
      mapping: PillType,
      control: 'select',
      table: {
        defaultValue: {
          summary: PillType.Primary
        }
      }
    },
    icon: {
      description: 'The icon component to be displayed',
      options: allIconsStoryOptions,
      control: 'select',
      table: {
        category: 'Icon',
        type: {
          summary: 'ReactElement | undefined'
        }
      }
    },
    isUppercase: {
      description: 'Use this property to uppercase the text content',
      control: 'boolean',
      table: {
        category: 'Styling'
      }
    },
    dataTestId: {
      control: 'text',
      description: 'Pill data-testid'
    },
    ...generateStorybookSpacing()
  }
};

type Story = StoryObj<PillProps>;

export const Base: Story = {
  args: {
    text: 'Text',
    icon: <CarlsbergFilled />,
    type: PillType.Primary,
    size: PillSize.Medium,
    isUppercase: false
  }
};

export const Icon: Story = {
  args: {
    ...Base.args,
    type: PillType.Success,
    text: ''
  }
};

export const Text: Story = {
  args: {
    ...Base.args,
    type: PillType.Fail,
    icon: undefined
  }
};

export default meta;
