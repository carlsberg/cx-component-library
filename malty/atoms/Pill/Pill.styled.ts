import styled, { css } from 'styled-components';
import { PillColor, PillSize } from './Pill.types';

export const StyledPill = styled.div<{
  size: string;
  fontSize: string;
  iconSize: string;
  padding: string;
  isRounded: boolean;
  hasOnClick: boolean;
  color: PillColor;
  hasText: boolean;
}>`
  font-family: inherit;
  font-size: ${({ fontSize }) => `${fontSize}`};
  line-height: 1;
  font-weight: bold;
  background-color: ${({ color, theme }) => theme.colors.colours.information[color].value};
  color: ${({ theme }) => theme.colors.colours.default.white.value};
  display: inline-flex;
  align-items: center;
  height: ${({ size }) => `${size}px`};
  transition: background-color 0.25s ease-in-out;

  ${({ hasOnClick }) =>
    hasOnClick &&
    css`
      cursor: pointer;
      &:hover {
        background: ${({ theme }) => theme.colors.colours.information.indirect.value};
      }
    `}
  ${({ isRounded, size }) =>
    isRounded &&
    css`
      border-radius: ${parseInt(size.replace('px', ''), 10) / 2}px;
    `}

  .pill {
    &__icon {
      margin-right: ${({ hasText, theme }) => hasText && theme.sizes['2xs'].value};
      height: ${({ iconSize }) => iconSize};
      width: ${({ iconSize }) => iconSize};
    }
    &__remove-icon {
      margin-left: ${({ padding }) => padding};
      height: ${({ iconSize }) => iconSize};
      width: ${({ iconSize }) => iconSize};
      cursor: pointer;
      transition: 0.25s ease-in-out;
      transition-property: fill, transform;
      &:hover {
        transform: rotate(90deg);
      }
    }
  }

  ${({ size, hasText, padding }) => {
    if (size === PillSize.ExtraSmall || size === PillSize.Small) {
      return css`
        padding: 0 ${padding};
        .pill {
          &__icon {
            ${hasText && `margin-right: ${parseInt(padding.replace('px', ''), 10) / 2}px;`};
          }
          &__remove-icon,
          &__icon {
            padding: ${parseInt(padding.replace('px', ''), 10) / 2}px 0;
            height: 100%;
            width: auto;
          }
        }
      `;
    }
    return css`
      padding: 0 ${padding}px;
    `;
  }}
`;
