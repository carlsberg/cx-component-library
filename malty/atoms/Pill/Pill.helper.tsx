import { IconColor, IconSize } from '@carlsberggbs/malty.atoms.base-icon';
import { defaultTheme } from '@carlsberggbs/malty.theme.new-malty-theme-provider';
import { useContext } from 'react';
import { ThemeContext } from 'styled-components';
import { IconTextColorProps, PillSize, PillType, UsePillBackgroundColorProps, UsePillStylesProps } from './Pill.types';

export const usePillStyles = ({ size }: UsePillStylesProps) => {
  const theme = useContext(ThemeContext) || defaultTheme;

  const pillStyles = {
    [PillSize.ExtraSmall]: {
      numSize: theme.sizesV2.s,
      fontWeight: theme.typographyV2.montserrat.body[10].semibold.fontWeight,
      lineHeight: theme.typographyV2.montserrat.body[10].semibold.lineHeight,
      fontSize: theme.typographyV2.montserrat.body[10].semibold.fontSize,
      iconSize: IconSize.ExtraSmall,
      padding: `${theme.sizesV2['4xs']} ${theme.sizesV2['3xs']}`,
      gap: theme.sizesV2['5xs']
    },
    [PillSize.Small]: {
      numSize: theme.sizesV2.m,
      fontWeight: theme.typographyV2.montserrat.body[10].semibold.fontWeight,
      lineHeight: theme.typographyV2.montserrat.body[10].semibold.lineHeight,
      fontSize: theme.typographyV2.montserrat.body[10].semibold.fontSize,
      iconSize: IconSize.Small,
      padding: `${theme.sizesV2['4xs']} ${theme.sizesV2['3xs']}`,
      gap: theme.sizesV2['4xs']
    },
    [PillSize.Medium]: {
      numSize: theme.sizesV2.xl,
      fontWeight: theme.typographyV2.montserrat.body[14].semibold.fontWeight,
      lineHeight: theme.typographyV2.montserrat.body[14].semibold.lineHeight,
      fontSize: theme.typographyV2.montserrat.body[14].semibold.fontSize,
      iconSize: IconSize.Medium,
      padding: `${theme.sizesV2['4xs']} ${theme.sizesV2['2xs']}`,
      gap: theme.sizesV2['4xs']
    }
  };

  return pillStyles[size];
};

// TODO: Refactor this when the icons are reviewed
export const useIconColor = ({ type }: IconTextColorProps) => {
  if (type === PillType.Archive || type === PillType.Success || type === PillType.Alert) {
    return IconColor.DigitalBlack;
  }

  if (type === PillType.Disabled) {
    return IconColor.Disabled;
  }

  return IconColor.White;
};

export const useTextColor = ({ type }: IconTextColorProps) => {
  const theme = useContext(ThemeContext) || defaultTheme;

  if (type === PillType.Archive || type === PillType.Success || type === PillType.Alert) {
    return theme.colorsV2.default.content[500];
  }

  if (type === PillType.Disabled) {
    return theme.colorsV2.default.foreground.primary['interactive--disabled'];
  }

  return theme.colorsV2.default.content[100];
};

export const usePillBackgroundColor = ({ type }: UsePillBackgroundColorProps) => {
  const theme = useContext(ThemeContext) || defaultTheme;

  switch (type) {
    case PillType.Success:
      return theme.colorsV2.default.success[500];
    case PillType.Alert:
      return theme.colorsV2.default.warning[500];
    case PillType.Fail:
      return theme.colorsV2.default.error[600];
    case PillType.Notification:
      return theme.colorsV2.default.information[600];
    case PillType.Archive:
      return theme.colorsV2.default.primary[200];
    case PillType.Secondary:
      return theme.colorsV2.default.primary[400];
    case PillType.Disabled:
      return theme.colorsV2.default.background.primary['interactive--disabled'];
    default:
      return theme.colorsV2.default.content[500];
  }
};
