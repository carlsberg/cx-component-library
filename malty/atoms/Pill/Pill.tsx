import { Icon, IconColor, IconSize } from '@carlsberggroup/malty.atoms.icon';
import { globalTheme as defaultTheme, TypographyProvider } from '@carlsberggroup/malty.theme.malty-theme-provider';
import React, { useContext, useEffect, useState } from 'react';
import { ThemeContext } from 'styled-components';
import { StyledPill, StyledText } from './Pill.styled';
import { PillColor, PillProps, PillSize } from './Pill.types';

export const Pill = ({ text, icon, iconColor, color = PillColor.Primary, size = PillSize.Medium }: PillProps) => {
  const theme = useContext(ThemeContext) || defaultTheme;
  const [numSize, setNumSize] = useState(theme.sizes.l.value);
  const [fontSize, setFontSize] = useState(theme.typography.desktop.text['medium-small_bold']['font-size'].value);
  const [iconSize, setIconSize] = useState(theme.sizes.m.value);
  const [padding, setPadding] = useState(theme.sizes.s.value);
  const [marginText, setMarginText] = useState(theme.sizes['4xs'].value);
  const [iconTextColor, setIconTextColor] = useState(iconColor || IconColor.White);

  useEffect(() => {
    switch (size) {
      case PillSize.ExtraSmall: {
        setNumSize(theme.sizes.s.value);
        setFontSize(theme.typography.desktop.text.micro_bold['font-size'].value);
        setIconSize(theme.sizes.xs.value);
        setPadding(theme.sizes['3xs'].value);
        setMarginText(theme.sizes['5xs'].value);
        break;
      }
      case PillSize.Small: {
        setNumSize(theme.sizes.m.value);
        setFontSize(theme.typography.desktop.text.tiny_bold['font-size'].value);
        setIconSize(theme.sizes.s.value);
        setPadding(theme.sizes.xs.value);
        setMarginText(theme.sizes['4xs'].value);
        break;
      }

      default: {
        setNumSize(theme.sizes.l.value);
        setFontSize(theme.typography.desktop.text['medium-small_bold']['font-size'].value);
        setIconSize(theme.sizes.m.value);
        setPadding(theme.sizes.s.value);
        setMarginText(theme.sizes['4xs'].value);
        break;
      }
    }
  }, [size, theme]);

  useEffect(() => {
    if (!iconColor && (color === PillColor.Archive || color === PillColor.Success || color === PillColor.alertStrong)) {
      setIconTextColor(IconColor.Primary);
    }
  }, [color, theme]);

  return (
    <TypographyProvider>
      <StyledPill
        color={color}
        size={numSize}
        fontSize={fontSize}
        iconSize={iconSize}
        padding={padding}
        hasText={!!text}
        theme={theme}
        textColor={iconTextColor}
        pillSize={size}
      >
        {icon && <Icon name={icon} size={IconSize.Small} color={iconTextColor} className="pill__icon" />}
        <StyledText theme={theme} marginText={marginText} hasText={!!text}>
          {text}
        </StyledText>
      </StyledPill>
    </TypographyProvider>
  );
};
