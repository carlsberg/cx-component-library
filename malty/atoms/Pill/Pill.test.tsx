import { CarlsbergFilled } from '@carlsberggroup/malty.icons.carlsberg-filled';
import { render } from '@carlsberggroup/malty.utils.test';
import { screen } from '@testing-library/react';
import React from 'react';
import { Pill } from './Pill';

describe('pill', () => {
  const text = 'Pill text';
  const iconDataTestId = 'carlsberg-icon';

  it('should render the component correctly', () => {
    render(<Pill text={text} icon={<CarlsbergFilled dataTestId={iconDataTestId} />} />);

    expect(screen.getByText(text)).toBeVisible();
    expect(screen.getByTestId(iconDataTestId)).toBeVisible();
  });

  it('should not render the component with the text and icon', () => {
    render(<Pill />);

    expect(screen.queryByText(text)).not.toBeInTheDocument();
    expect(screen.queryByTestId(iconDataTestId)).not.toBeInTheDocument();
  });

  it('should render only text', () => {
    render(<Pill text={text} />);

    expect(screen.getByText(text)).toBeVisible();
    expect(screen.queryByTestId(iconDataTestId)).not.toBeInTheDocument();
  });

  it('should render only Icon', () => {
    render(<Pill icon={<CarlsbergFilled dataTestId={iconDataTestId} />} />);

    expect(screen.queryByText(text)).not.toBeInTheDocument();
    expect(screen.getByTestId(iconDataTestId)).toBeVisible();
  });
});
