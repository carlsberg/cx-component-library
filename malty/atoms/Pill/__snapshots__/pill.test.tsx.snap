// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pill matches snapshot 1`] = `
<span
  className="sc-bczRLJ hROJNc"
>
  <div
    className="sc-dkzDqf hoMrIK"
    color="primary"
    fontSize="14px"
    size="32px"
  >
    <svg
      className="sc-gsnTZi iOMLMs pill__icon"
      color="white"
      data-testid="svg-component"
      size="24px"
      version="1.1"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <path
        d="M12 6.753a1.33 1.33 0 01-1.341-1.32V2.757C10.659 1.79 9.867 1 8.877 1c-.41 0-.784.153-1.087.388a4.256 4.256 0 01-4.149.61C2.433 1.533 1.023 1.981.353 3.124a2.618 2.618 0 00.65 3.372 4.12 4.12 0 011.542 3.842c-.055.38-.002.774.204 1.13a1.801 1.801 0 002.433.64l2.355-1.34a1.348 1.348 0 011.826.486 1.31 1.31 0 01-.495 1.8l-2.35 1.339a1.737 1.737 0 00-.646 2.398c.204.355.53.594.883.734 1.375.546 2.387 1.76 2.608 3.24A2.653 2.653 0 0012 23c1.34 0 2.438-.967 2.629-2.234.229-1.48 1.243-2.695 2.611-3.241.36-.14.682-.38.887-.734a1.736 1.736 0 00-.654-2.398l-2.351-1.34a1.305 1.305 0 01-.486-1.8c.369-.63 1.187-.85 1.824-.486l2.35 1.342a1.809 1.809 0 002.44-.642c.2-.355.256-.75.2-1.129a4.117 4.117 0 011.538-3.842c1.014-.8 1.327-2.226.654-3.372-.672-1.143-2.073-1.591-3.288-1.126a4.252 4.252 0 01-4.148-.61 1.765 1.765 0 00-2.87 1.37v2.674c0 .728-.6 1.32-1.336 1.32"
        fillRule="evenodd"
      />
    </svg>
    <div
      className="sc-hKMtZM dKAPlu"
      data-testid="undefined-text"
    >
      Pill text
    </div>
  </div>
</span>
`;
